hazelcast-sql/src/test/java/com/hazelcast/jet/sql/SqlClientCompactQueryTest.java

```diff
@@ -0,0 +1,324 @@
+        byte[] bytes = {1, 0};
+        short[] shorts = {13, 23};
+        char[] chars = {'a', 'b'};
+        int[] ints = {321, 421};
+        long[] longs = {3231L, 1513L};
+        float[] floats = {32123.231f, 4343.32f};
+        double[] doubles = {3212.3, 413232.1};
+        BigDecimal[] decimals = {BigDecimal.valueOf(3213), BigDecimal.valueOf(313)};
+        LocalTime[] times = {LocalTime.now()};
+        LocalDate[] dates = {LocalDate.now()};
+        LocalDateTime[] tmstmps = {LocalDateTime.now()};
+        OffsetDateTime[] tmstmptzs = {OffsetDateTime.now()};
+        String[] strings = {"hello", "world"};
+        boolean bool = true;
+        byte b = (byte) 1;
+        short s = (short) 13;
+        char c = 'a';
+        int i = 321;
+        long l = 31231L;
+        float f = 32123.231f;
+        double d = 3212.3;
+        BigDecimal dl = BigDecimal.valueOf(3213);
+        String st = "test";
+        LocalTime t = LocalTime.now();
+        LocalDate dt = LocalDate.now();
+        LocalDateTime tmstmp = LocalDateTime.now();
+        OffsetDateTime tmstmptz = OffsetDateTime.now();
+
+        client.getSql().execute("CREATE MAPPING " + "test" + '('
+                + "__key INT "
+                + ", bool OBJECT "
+                + ", b OBJECT"
+                + ", s OBJECT"
+                + ", c OBJECT"
+                + ", i OBJECT"
+                + ", l OBJECT"
+                + ", f OBJECT"
+                + ", d OBJECT"
+                + ", st OBJECT"
+                + ", dl OBJECT"
+                + ", t OBJECT"
+                + ", dt OBJECT"
+                + ", tmstmp OBJECT"
+                + ", tmstmptz OBJECT"
+                + ", booleans OBJECT "
+                + ", bytes OBJECT"
+                + ", shorts OBJECT"
+                + ", chars OBJECT"
+                + ", ints OBJECT"
+                + ", longs OBJECT"
+                + ", floats OBJECT"
+                + ", doubles OBJECT"
+                + ", strings OBJECT"
+                + ", decimals OBJECT"
+                + ", times OBJECT"
+                + ", dates OBJECT"
+                + ", tmstmps OBJECT"
+                + ", tmstmptzs OBJECT"
+                + ") TYPE " + IMapSqlConnector.TYPE_NAME + ' '
+                + "OPTIONS ("
+                + '\'' + OPTION_KEY_FORMAT + "'='" + "int" + '\''
+                + ", '" + OPTION_VALUE_FORMAT + "'='" + COMPACT_FORMAT + '\''
+                + ", '" + OPTION_VALUE_COMPACT_TYPE_NAME + "'='" + "testObject" + '\''
+                + ")"
+        );
+
+        client.getSql().execute("SINK INTO test "
+                        + "( __key, bool, b, s, c, i, l, f, d, st, dl, t, dt, tmstmp, tmstmptz"
+                        + ", booleans, bytes, shorts, chars, ints, longs, floats, doubles, strings,"
+                        + " decimals, times, dates, tmstmps, tmstmptzs)"
+                        + " VALUES (? "
+                        + ",CAST (? AS BOOLEAN)"
```

Comment:
I did not like the way that we have to cast the primitives but not the arrays when using. I was expecting all behave the same way. Can this be a backward compatibility issue when the array types are introduced later on?